src/otp-input/otp-input.tsx

```diff
@@ -118,62 +114,96 @@ export const OtpInput = ({
 
     const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
         setOtpValues(new Array(numOfInput).fill(""));
-        setEnableCountDownTimer(true);
+
+        if (!isWithinCooldown()) {
+            setLastCtaTimestamp(new Date());
+            setCountDown(cooldownDuration);
+        }
+
         if (onClick) {
             onClick(event);
         }
     };
 
+    // =========================================================================
+    // HELPER FUNCTIONS
+    // =========================================================================
+    const validateUserInput = (value: string, length = 1) =>
+        !value ? false : RegExp(`^[0-9]{${length}}$`).test(value);
+
+    const isWithinCooldown = (): boolean => {
+        const currentTime = Date.now();
+        const coolDownInMilliseconds = cooldownDuration * 1000;
+
+        return (
+            currentTime < lastCtaTimestamp.valueOf() + coolDownInMilliseconds
+        );
+    };
+
+    const beginCountdown = () => {
+        let timer = cooldownDuration;
+        const interval = setInterval(() => {
+            timer--;
+            setCountDown(timer);
+            if (timer === 0) {
```

Comment:
1. calculate remaining seconds left with `Math.ceil((lastCtaTimestamp.valueOf() + cooldownDuration * 1000 - Date.now()) / 1000)`
2. use that for the countdown instead
3. when remaining seconds <= 0, clear interval and trigger `onCooldownEnd`

mainly to keep `onCooldownEnd` synced as much as possible